apps/dashboard/src/components/workflow-editor/configure-workflow.tsx

```diff
@@ -0,0 +1,137 @@
+              <FormControl>
+                <InputField>
+                  <Input placeholder="Untitled" {...field} />
+                </InputField>
+              </FormControl>
+              <FormMessage />
+            </FormItem>
+          )}
+        />
+        <FormField
+          control={control}
+          name="workflowId"
+          render={({ field }) => (
+            <FormItem>
+              <FormLabel>Workflow Identifier</FormLabel>
+              <FormControl>
+                <InputField className="flex overflow-hidden pr-0">
+                  <Input placeholder="Untitled" {...field} />
+                  <CopyButton
+                    content={field.value}
+                    className="rounded-md rounded-s-none border-b-0 border-r-0 border-t-0 text-neutral-400"
+                  />
+                </InputField>
+              </FormControl>
+              <FormMessage />
+            </FormItem>
+          )}
+        />
+        <FormField
+          control={control}
+          name="description"
+          render={({ field }) => (
+            <FormItem>
+              <FormLabel>Description</FormLabel>
+              <FormControl>
+                <Textarea placeholder="Untitled" {...field} />
+              </FormControl>
+              <FormMessage />
+            </FormItem>
+          )}
+        />
+        <FormField
+          control={control}
+          name="tags"
+          render={({ field }) => (
+            <FormItem>
+              <div className="flex items-center gap-1">
+                <FormLabel>Add tags</FormLabel>
+              </div>
+              <FormControl className="text-xs text-neutral-600">
+                <TagInput {...field} value={field.value ?? []} showAddButton />
+              </FormControl>
+            </FormItem>
+          )}
+        />
+      </div>
+      <Separator />
+      <FormField
+        control={control}
+        name="preferences.default.all.readOnly"
+        render={({ field }) => (
+          <FormItem className="flex items-center justify-between gap-2.5 space-y-0 px-3 py-4">
+            <div className="flex items-center gap-1">
+              <FormLabel>Mark as critical</FormLabel>
+              <TooltipProvider>
+                <Tooltip>
+                  <TooltipTrigger asChild>
+                    <div>
+                      <RiInformation2Line className="text-foreground-400 cursor-pointer" />
+                    </div>
+                  </TooltipTrigger>
+                  <TooltipContent className="w-72">
+                    <span>Marking this workflow as critical will prevent subscribers from unsubscribing.</span>
+                  </TooltipContent>
+                </Tooltip>
+              </TooltipProvider>
+            </div>
+            <FormControl>
+              <Switch checked={field.value} onCheckedChange={field.onChange} />
+            </FormControl>
+          </FormItem>
+        )}
+      />
```

Comment:
There have been some changes in the designs around the preferences. Let's remove this field and add the new "Configure preferences" button (but let's keep the schema), and we will do that in a separate ticket. Please add the placeholder ticket and ask me to fill in the rest.